### PR TITLE
Add SidekiqStatistics middeware

### PR DESCRIPTION
**Pull request by gonzalorey**
## Description
This logic is already in **rcore** [here](https://github.com/restorando/rcore/blob/master/lib/r_core/job_statistics.rb), and I intend to add it to **barenboim** to track job statistics with **Sidekiq**, therefore I thought of moving it here and avoid repeating all this logic again. 

In this PR https://github.com/restorando/barenboim/pull/249 it's clear how little it's required to start reporting **Sidekiq**'s metrics.

## Tasks

- [x] Add jobs tracking logic
- [x] Add specs
- [x] Update CHANGELOG.md & bump version

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.5.1
```
[the remark this answers is not in the document]
Comment (gonzalorey) [reply .ruby-version]: Actually, not really sure if we should keep this parameter here or not, there's nothing telling which ruby version was used for this gem, and I just added one that seemed to fit.

File: statue.gemspec [hunk 2]
```diff
@@ -22,4 +22,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "rack", "~> 1.6"
   spec.add_development_dependency "rack-test", "~> 0.6.3"
   spec.add_development_dependency "minitest"
+  spec.add_development_dependency "sidekiq"
+  spec.add_development_dependency "pry-rescue"
```
Comment (gonzalorey) [inline statue.gemspec]: **pry-rescue** really helped me while fixing the specs, so I decided to leave it as a development dependency (open to discussion though).

File: test/sidekiq_statistics_test.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      msg["retry_count"] = 1
+      middleware.call(worker, msg, msg["queue"]) { true }
+
+      retry_metric = metrics.find { |m| m.name == "job.default.FooJob.retry" }
+      assert_equal(retry_metric.type, :c)
+    end
+
+    it "tracks queue latency" do
+      # Time.now - 1 minute
+      msg["enqueued_at"] = Time.now - 60
```
Comment (gonzalorey) [inline test/sidekiq_statistics_test.rb]: Without Rails this was my best shot!

File: statue.gemspec [hunk 4]
```diff
@@ -22,4 +22,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "rack", "~> 1.6"
   spec.add_development_dependency "rack-test", "~> 0.6.3"
   spec.add_development_dependency "minitest"
+  spec.add_development_dependency "sidekiq", "~> 3.2"
+  # spec.add_development_dependency "pry-rescue"
```
Comment (gonzalorey) [inline statue.gemspec]: Estos los agregué porque simplificaban bastante el debugging con **Minitest**, pero se ve que no se bancan el build con **jruby**...

File: lib/statue/sidekiq_statistics.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  #   only if the job was throttled by sidekiq-throttler
+  # * count job.<queue>.<job_name>.overdue:
+  #   only if the deadline from run_deadline_middleware was reached
+  # * count job.<queue>.<job_name>.retry:
+  #   only if the job corresponds to a retry for a previously failed job
+  # * duration job.<queue>.<job_name>.latency:
+  #   time difference between now and when the job last entered the queue
+  # * duration job.<queue>.<job_name>:
+  #   job run duration (reported only if the job doesn't fail)
+  module SidekiqStatistics
```
Comment (gonzalorey) [inline lib/statue/sidekiq_statistics.rb]: Les copa este nombre? O prefieren `JobStatistics`?

Comment (gonzalorey) [reply lib/statue/sidekiq_statistics.rb]: This middleware hooks itself to a **Sidekiq**'s middleware, not sure if other job runners are compatible.

File: .ruby-version (hunk 1 above)
[the remark this answers is not in the document]
Comment (gonzalorey) [reply .ruby-version]: Ended up using locally `2.2.0`, but I'm not sure if this should be added or not to the repo. I think I'd better remove it, as it's more related to local configurations.

File: statue.gemspec (hunk 4 above)
Comment (gonzalorey) [reply statue.gemspec]: 🐩

File: test/sidekiq_statistics_test.rb (hunk 3 above)
Comment (kingpijamas) [reply test/sidekiq_statistics_test.rb]: Aren't we using active support here? 😨

Comment (gonzalorey) [reply test/sidekiq_statistics_test.rb]: Who wants Rails or anything like that!?

**Pull request merged**